pkg/controller/managedresource/add.go

```diff
@@ -140,3 +153,41 @@ func (o *ControllerOptions) Completed() *ControllerConfig {
 func (c *ControllerConfig) ApplyClassFilter(filter *filter.ClassFilter) {
 	*filter = *c.ClassFilter
 }
+
+// ApplyDefaultClusterId sets completes the cluster id according to a dedicated cluster access
+func (c *ControllerConfig) ApplyDefaultClusterId(log logr.Logger, restcfg *rest.Config) error {
+	if c.ClusterID == "<cluster>" || c.ClusterID == "<default>" {
+		log.Info("Trying to get cluster id from cluster")
+		tmpClient, err := client.New(restcfg, client.Options{})
+		if err == nil {
+			c.ClusterID, err = determineClusterIdentity(tmpClient, c.ClusterID == "<cluster>")
+		}
+		if err != nil {
+			return fmt.Errorf("unable to determine cluster id: %+v", err)
+		}
+	}
+	return nil
+}
+
+// determineClusterIdentity is used to extract the cluster identity from the cluster-identity
+// config map. This is intended as fallback if no explicit cluster identity is given.
+// in  seed-shoot scenario, the cluster id for the managed resources must be explicitly given
+// to support the migration of a shoot from one seed to another. Here the identity `seed` should
+// be set.
+func determineClusterIdentity(c client.Client, force bool) (string, error) {
+	cm := corev1.ConfigMap{}
+	err := c.Get(context.Background(), client.ObjectKey{Name: gardenerconstantsv1beta1.ClusterIdentity, Namespace: metav1.NamespaceSystem}, &cm)
+	if err == nil {
+		if id, ok := cm.Data[gardenerconstantsv1beta1.ClusterIdentity]; ok {
+			return id, nil
+		}
+		if force {
+			return "", fmt.Errorf("cannot determine cluster identity from configmap: no cluster-identity entry ")
+		}
+	} else {
+		if force {
+			return "", fmt.Errorf("cannot determine cluster identity from configmap: %s", err)
+		}
```

Comment:
If `force=false` then this also ignores error other than "not found". Shouldn't we only ignore it if `apierrors.IsNotFound(err) == true`?

Reply:
yepp